packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -20,14 +21,18 @@ export class CheckboxBase extends React.Component<ICheckboxProps, ICheckboxState
   private _id: string;
   private _classNames: { [key in keyof ICheckboxStyles]: string };
 
-  public static getDerivedStateFromProps(props: ICheckboxProps, state: ICheckboxState): ICheckboxState {
+  public static getDerivedStateFromProps(props: ICheckboxProps, state: ICheckboxState): ICheckboxState | null {
+    if (!props.defaultIndeterminate && state.isIndeterminate) {
```

Comment:
You're checking falsey values here. Shouldn't you be checking for an `undefined` exact match?

```tsx
if (props.indeterminate !== undefined) {
  return { isIndeterminate: props.indeterminate };
}
```

Reply:
also, it's ignoring `props.checked` in this codepath...

Perhaps a better approach would be:

```tsx
const { 
  indeterminate = state.indeterminate,
  checked = state.checked
 } = props;

return {
  checked,
  indeterminate
};
```

Reply:
This way, if props contains a controlled value for indeterminate or checked, it's always preferred over whatever is currently in the state.

Reply:
I think there's a confusing scenario where both indeterminate and checked are provided, which seems conflicting.